### PR TITLE
RFC-0001: Add spans for referring to individual values and user goals

### PR DESCRIPTION
**Pull request by twelho**
This enables us to refer to these from other RFCs without relying on the numbering.

### CODE REVIEW

File: docs/rfcs/0001-high-level-architecture.md [hunk 1]
```diff
@@ -112,64 +112,64 @@ The following values apply to the whole system, and are sorted roughly in priori
 
 > **Disclaimer:** All of these values are aspirational, they are not literal guarantees that can be used for liability claims. We expect to iteratively improve towards and get closer to these goals as the project matures and new versions are released.
 
-1. **Security**
+1. <span id="value-security"></span> **Security**
    1. **Security by design**: Security should be at the top of our minds at every decision we make. All design proposals must consider how the proposed change affects security concerns.
```
Comment (luxas) [inline docs/rfcs/0001-high-level-architecture.md:116]: I would also add spans to all of these; IMO they are important to reference separately as well

Comment (twelho) [reply docs/rfcs/0001-high-level-architecture.md]: Sure, it's going add a bit of clutter but I can do that.

Comment (twelho) [reply docs/rfcs/0001-high-level-architecture.md]: Done

File: docs/rfcs/0001-high-level-architecture.md [hunk 2]
```diff
@@ -23,9 +23,9 @@
 
 **Creation Date**: `2020-12-10`
 
-**Last Updated**: `2020-12-10`
+**Last Updated**: `2020-06-08`
 
-**Version Number**: `v1.0.1`
+**Version Number**: `v1.1.0`
```
Comment (luxas) [inline docs/rfcs/0001-high-level-architecture.md:28]: this much was a bit unexpected maybe, but I'm fine with it I guess, as most of the changes (e.g. removing two values) were already done in v1.0.1

Comment (twelho) [reply docs/rfcs/0001-high-level-architecture.md:28]: Yep, the previous PR was a bit rushed (last updated date wasn't updated either), so with all the changes this should already be v1.1.0.

**Pull request merged**